ddtrace/internal/_encoding.pyx

```diff
@@ -101,7 +101,7 @@ cdef inline int pack_number(msgpack_packer *pk, object n):
     raise TypeError("Unhandled numeric type: %r" % type(n))
 
 
-cdef inline int pack_text(msgpack_packer *pk, object text):
```

Comment:
Not 100% yet that this exception handling is required, going to remove and confirm

Reply:
Yeah so for `cdef`s that declare a return type (that isn't a python type, which is implicit) then exceptions are not automatically raised and get logged instead.

```cython
cdef int c_division(x, y):
    if y == 0:
        raise ZeroDivisionError()
    return x / y

cpdef div(x, y, reraise):
        return c_division(x, y)
```

```python
>>> div(1, 0)
Exception ZeroDivisionError: ZeroDivisionError() in 'ddtrace.internal._rand.c_division' ignored
```

Reply:
This is documented here: https://notes-on-cython.readthedocs.io/en/latest/misc.html#cdef-exceptions-and-the-return-type

Reply:
Good find!